### DISCUSSION TITLE
Certificate expired for Machine ID proxy

### DISCUSSION THREAD

**Discussion by Aloysius-AS** [gravitational/teleport#33195] [Q&A] [answered]
Hello, I am referencing [Machine ID with Databases](https://goteleport.com/docs/machine-id/guides/databases/#step-23-configure-and-start-machine-id) and created a proxy for an API application in a systemd.unit file like so:
```shell
ExecStart=/usr/local/bin/tbot -c /etc/tbot.yaml proxy --proxy=proxy.example.com:443 --destination-dir=/opt/machine-id/rest-api-app app --port=9000 rest-api-app
``` 
The client application connects to the API application via http://127.0.0.1:9000.
After an hour, I noticed that the certificate used by the proxy expired.
```shell
2023-10-10T05:53:50Z WARN [ALPN:PROX] Failed to handle client connection. error:[
ERROR REPORT:
Original Error: *tls.CertificateVerificationError tls: failed to verify certificate: x509: certificate has expired or is not yet valid: current time 2023-10-10T05:53:50Z is after 2023-10-10T04:48:14Z
Stack Trace:
         github.com/gravitational/teleport/lib/srv/alpnproxy/proxy.go:392 github.com/gravitational/teleport/lib/srv/alpnproxy.(*Proxy).handleConn
         github.com/gravitational/teleport/lib/srv/alpnproxy/proxy.go:326 github.com/gravitational/teleport/lib/srv/alpnproxy.(*Proxy).Serve.func1
         runtime/asm_amd64.s:1598 runtime.goexit
User Message: tls: failed to verify certificate: x509: certificate has expired or is not yet valid: current time 2023-10-10T05:53:50Z is after 2023-10-10T04:48:14Z] alpnproxy/proxy.go:337
``` 

I am guessing this is caused by the proxy using the certificate that was generated at the point of time when the service was started.

The error resolves after the service is restarted, but appears again after an hour. I am using Teleport v13.1.0.

Is that any way for the proxy to automatically use the renewed certificates in `destination-dir`?

**Comment by webvictim** [comment, marked as answer]
This is a bit of a gap currently as the client doesn't automatically re-read certificates on disk when they're changed. For now, you could write such logic yourself (using `inotify` or similar) or just restart your client periodically.

**Reply by Aloysius-AS** [reply]
Thank you. I'll explore that.
